### DISCUSSION TITLE
Privacy concern of `Code Snippets Data Collection`

### DISCUSSION THREAD

**Discussion by garylip** [community/community#52511] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi team, may I know if Copilot also reads other files and resources in the project even if we only enable it for some specific file types?

Based on the privacy term, it should not, am I right?
<img width="934" alt="Screenshot 2023-04-11 at 6 39 40 PM" src="https://user-images.githubusercontent.com/28523022/231136374-9bf552d0-8cf2-4b8f-ae2c-e2381c69c6ab.png">

**Comment by davecheney** [comment, marked as answer]
Copilot for Business may consult adjacent open tabs, or other files on disk when constructing the prompt. As above, for Copilot for Business, no code snippets are retained following the request.

**Reply by ultrabstrong** [reply]
Hi @davecheney , is this additional data only used locally to construct the prompt or it get sent to the server? Also, does "prompt" mean the same thing as "code snippet" for this?

**Reply by davecheney** [reply]
additional data is used when constructing the prompt which is sent to the model to generate a completion response. prompts, and completion responses, are considered code snippets.

**Reply by darrenyeats** [reply]
Does the same apply to Copilot Labs in terms of not retaining snippets? Thanks.

**Reply by davecheney** [reply]
Copilot labs is bound by its own terms of service.

**Reply by darrenyeats** [reply]
Thanks David! Searched for terms of service - some more explicit info here https://github.com/orgs/community/discussions/52767#discussioncomment-5617523
